packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -636,10 +645,18 @@ class _Label extends StatelessWidget {
       );
     }
 
-    return Align(
-      alignment: Alignment.bottomCenter,
-      heightFactor: 1.0,
-      child: Container(child: text),
+    final MediaQueryData mediaQueryData = MediaQuery.of(context);
+    return MediaQuery(
+      // Do not grow text in bottom navigation bar because there is no room,
+      // instead, properly scaled tooltips will show on long press.
+      data: mediaQueryData.copyWith(
+        textScaleFactor: math.min(1.0, mediaQueryData.textScaleFactor),
```

Comment:
I hadn't really thought about this before in other contexts, but is there any advantage in scaling down the text if it already fits?

Reply:
In this particular case, I believe that there are some users who are using `textScaleFactor: 0` as a workaround to hide the labels, I do not want to break this use case. There was a period of time when `BottomNavigationBar` did not support hiding labels on the items.